src/python/WMCore/Services/MonIT/Grafana.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+# coding=utf-8
+"""
+Wrapper class, based on the PyCuRL module, providing an interface
+to CERN MonIT Grafana APIs
+"""
+from __future__ import division, print_function, absolute_import
+
+import logging
+import json
+from copy import copy
+from pprint import pformat
+from WMCore.Services.pycurl_manager import RequestHandler
+try:
+    from urlparse import urljoin
+except ImportError:
+    # PY3
+    from urllib.parse import urljoin
+
+class Grafana(object):
+    """
+    Service class providing functionality to CERN monitoring Grafana APIs
+    """
+
+    def __init__(self, token, configDict=None):
+        """
+        Constructs a Grafana object.
+        :param token: mandatory string with the key/token string
+        :param configDict: dictionary with extra parameters, such as:
+          logger: logger object
+          endpoint: string with the url/endpoint to be used
+          cacheduration: float with the cache duration, in hours
+          headers: dictionary with the headers to be used
+        """
+        self._token = token
+        self.configDict = configDict or {}
+        self.configDict.setdefault('endpoint', "https://monit-grafana.cern.ch")
+        self.configDict.setdefault('cacheduration', 0)  # in hours
+        if 'headers' not in self.configDict:
+            self.configDict.setdefault('headers', {})
+            self.configDict['headers'].update({"Accept": "application/json"})
+            self.configDict['headers'].update({"Content-Type": "application/json"})
+
+        self.logger = self.configDict.get("logger", logging.getLogger())
+        self.logger.info("MonIT service initialized with parameters: %s", self.configDict)
+
+    def updateToken(self, token):
```

Comment:
Where is that one called? What should be the frequency of token refreshments? Is it supposed to be an external service which will take care of that?

Reply:
It's not called anywhere, yet. However, it might be needed in the future (if we keep using this module)